lib/Sema/CSSimplify.cpp

```diff
@@ -5947,8 +5930,27 @@ bool ConstraintSystem::repairFailures(
       }
     }
 
+    FixBehavior fixBehavior = FixBehavior::Error;
+
+    if (isForCodeCompletion()) {
+      // If the argument contains the code completion location, the user has not
+      // finished typing out this argument yet. Treat the mismatch as valid so
+      // we don't penalize this solution.
+      if (auto *arg = getAsExpr(simplifyLocatorToAnchor(loc))) {
+        // Ignore synthesized args like $match in implicit pattern match
+        // operator calls. Their source location is usually the same as the
+        // other (explicit) argument's so source range containment alone isn't
+        // sufficient.
+        bool isSynthesizedArg = arg->isImplicit() && isa<DeclRefExpr>(arg);
+        if (!isSynthesizedArg && isForCodeCompletion() &&
+            containsIDEInspectionTarget(arg) && !lhs->isVoid() &&
+            !lhs->isUninhabited())
+          fixBehavior = FixBehavior::Suppress;
```

Comment:
What happens if this returns `true` like before instead of trying to suppress the diagnostic which wouldn't be emitted anyway in the code completion mode?

Reply:
The reason I want to record the fix is so I can inspect it while producing code completion results here: https://github.com/apple/swift/pull/72568/files#diff-1e522eaf116c0d9062c2ba193e9a4cd86687f87ebd6d3afbd705917a2f5746dbR178-R192. 

That’s basically the content of https://github.com/apple/swift/pull/72568 which this PR depends on.

Reply:
Hm, `containsIDEInspectionTarget` is overly broad then... If there is a mismatch between result type of a call where code completion token is used and some contextual type, I think presence of completion token shouldn't affect that.

Is the following example covered by this check as well:

```
func other(_: Int) {}
func test(_: String) {}
func compute(_: Int) -> String {}

other(test(compute(#^COMPLETE^#)))
```